### PR TITLE
Feature/fix gradle4 warning: Gradle now uses separate output directories

### LINKED ISSUE #168: Gradle 4 warning: Gradle now uses separate output directories for each JVM language...
Run with gradle-tomcat-plugin, gradle warning: 

Gradle now uses separate output directories for each JVM language, but this build assumes a single directory for all classes from a source set. This behaviour has been deprecated and is scheduled to be removed in Gradle 5.0

in this file: plugin/src/main/groovy/com/bmuschko/gradle/tomcat/tasks/TomcatRun.groovy

method: 

```
private void setupClassesJarScanning() {
    ....
    File resolvedClassesDirectory = getClassesDirectory()
    ....
}

```

CALL getClassesDirectory() will print those warning.

### PR DESCRIPTION
**Pull request by xingxiuyi**
fix #168

simple return sourceSets.main.output.classesDirs first element.

### CODE REVIEW

File: plugin/src/main/groovy/com/bmuschko/gradle/tomcat/TomcatBasePlugin.groovy [hunk 1]
```diff
@@ -57,7 +58,10 @@ class TomcatBasePlugin implements Plugin<Project> {
                 File webAppDir = getWarConvention(project).webAppDir
                 webAppDir.exists() ? webAppDir : null
             }
-            conventionMapping.map('classesDirectory') { project.sourceSets.main.output.classesDir.exists() ? project.sourceSets.main.output.classesDir : null }
+            conventionMapping.map('classesDirectory') {
```
Comment (bmuschko) [inline plugin/src/main/groovy/com/bmuschko/gradle/tomcat/TomcatBasePlugin.groovy]: For this change we will have to consider all classes directories. Just picking the first one won't do. Could you please change the field in `TomcatRun` to use a `FileCollection` instead? Any handling of the field in the task needs to change accordingly.

Comment (xingxiuyi) [reply plugin/src/main/groovy/com/bmuschko/gradle/tomcat/TomcatBasePlugin.groovy]: ok

**Comment by bmuschko** [comment]
@xingxiuyi Are you still interested in finishing off this pull request?

**Comment by bmuschko** [comment]
I am going to close this one as it is not yet mergeable. I will probably have a look at it myself to get rid of the deprecation message. Thanks again for the pull request.

**Comment by bmuschko** [comment]
I released version 2.5 which fixes the problem.

**Pull request closed without merging**